Chat App/Services/Implementation/UserManagementService.swift

```diff
@@ -35,24 +38,24 @@ extension UserManagementService {
      */
     func userSignin(username: String, hash: String) {
         task.enter()
-        print("Retrieving: \(username)")
         userHash = hash
         reference?.whereField(Constants.FirebaseStrings.userReference, isEqualTo: username)
             .getDocuments() { (snapshot, error) in
                 if let err = error {
-                    print("Error: \(err)")
+                    print("Error: \(err.localizedDescription)")
                     self.hasSignInFailed = true
                 } else {
-                    for document in snapshot!.documents {
-                        let data = document.data()
-                        self.username = (data["username"] as? String)!
-                        self.receivedHash = (data["password"] as? String)!
-                        self.receivedUUID = (document.documentID)
-                        self.isUserFound = true
-                    }
+                    if snapshot?.documents.count == 1 {
+                        guard let document = snapshot!.documents.first else { return }
```

Comment:
remove force cast since it will be wrapped to `document` variable

Reply:
changed force downcast to conditional downcast.